platforms/core-configuration/configuration-cache/src/main/kotlin/org/gradle/configurationcache/problems/ConfigurationCacheReport.kt

```diff
@@ -158,9 +163,11 @@ class ConfigurationCacheReport(
             fun ManagedExecutor.shutdownAndAwaitTermination() {
                 shutdown()
                 if (!awaitTermination(30, TimeUnit.SECONDS)) {
```

Comment:
❓ I don't think we should double the waiting time if saving is stuck. We can reduce it to 1-5 seconds, as the executor shouldn't have any work items to complete on a happy path, just shut down the thread.

Reply:
Done